thorlcr/thorutil/thbuf.cpp

```diff
@@ -2549,6 +2543,9 @@ class CSharedFullSpillingWriteAhead : public CInterfaceOf<ISharedRowStreamReader
         outputStream->flush();
         totalInputRowsRead.fetch_add(newRowsWritten);
         tempFileOwner->noteSize(iFileIO->getStatistic(StSizeDiskWrite));
+        CRuntimeStatisticCollection currentFileStats(inactiveStats.queryMapping());
+        ::mergeStats(currentFileStats, iFileIO);
```

Comment:
I'm not sure this is right.
currenFileStats has a spillingWriteAheadStatistics mapping, those stats are not within iFileIO ..

I think you either need to store as a mapping based on diskLocalStatistic, and allow the caller to remap (that's what the splitter code is doing), or, remap here and have no remapping in the caller.
It is probably better to accrue the stats in this container unmapped, and only remap in gatherStatistics.